BrowserKit/Sources/Redux/Action.swift

```diff
@@ -10,14 +10,22 @@ import Foundation
 /// action, including a UUID for a particular window. If an Action is truly global in
 /// nature or can't be associated with a UUID it can send either the UUID for the active
 /// window (see `WindowManager.activeWindow`) or if needed `WindowUUID.unavailable`.
-public protocol Action {
-    var windowUUID: UUID { get }
-}
+open class Action {
+    public var windowUUID: UUID
+    public var viewUUID: UUID?
+    public var actionType: ActionType
+
+    public init(windowUUID: UUID, viewUUID: UUID? = nil, actionType: ActionType) {
+        self.windowUUID = windowUUID
+        self.viewUUID = viewUUID
+        self.actionType = actionType
+    }
 
-extension Action {
     func displayString() -> String {
         let className = String(describing: Self.self)
         let actionName = String(describing: self).prefix(20)
         return "\(className).\(actionName)"
     }
 }
+
+public protocol ActionType {}
```

Comment:
I believe it would be beneficial to use enums for actions, mainly for two reasons. Firstly, the autocomplete feature in Xcode significantly enhances action discoverability. It conveniently allows you to view all options in one place. Secondly, using enums helps ensure that every new action added is handled properly because the compiler will flag an error if any action is overlooked. This approach helps avoid the potential oversight that might occur with a default case in switch statements.

Reply:
The objects that conform to ActionType are enums.

Reply:
When you switch on ActionType, you loose this capability.

Reply:
I may not be understanding. Are you proposing we have one big ActionType enum for all actions in the project?